src/libraries/Hooks.sol

```diff
@@ -98,4 +103,8 @@ library Hooks {
     function shouldCallAfterDonate(IHooks self) internal pure returns (bool) {
         return uint256(uint160(address(self))) & AFTER_DONATE_FLAG != 0;
     }
+
+    function shouldAccessLock(IHooks self) internal pure returns (bool) {
```

Comment:
As much as its named in line with the others, I think this name doesnt feel right. Its not that the hook "should" access the lock, its that it "can" access the lock.

I think for this and for NoOp, we should either go with `canAccessLock` and `canNoOp` or `shouldAllowAccessLock` and `shouldAllowNoOp`. I actually think "can" makes the most sense but "should allow" is in line with the shoulds haha

Reply:
Alice and I talked about using `hasPermissionToAccessLock` and `hasPermissionToNoOp`.. Thoughts?